tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -1840,11 +1856,21 @@ class CudnnRNNBackwardOp<GPUDevice, T> : public CudnnRNNKernelCommon {
     const Tensor* params = nullptr;
     const Tensor* sequence_lengths = nullptr;
     CudnnRnnModelShapes model_shapes;
+    bool use_padded_io = false;
     if (var_seq_lengths) {
       OP_REQUIRES_OK(context, ExtractForwardInput(
                                   context, model_types(), time_major, &input,
                                   &input_h, &input_c, &params,
                                   &sequence_lengths, num_proj, &model_shapes));
+      auto seq_array = sequence_lengths->template flat<int>().data();
```

Comment:
Can we have a helper function ShouldUsePaddedIO() to encapsulate and reuse this logic?

Reply:
Fixed. (And I added more description on the numbers of extra mem allocation.) Thanks.